dockstore-integration-testing/src/test/java/io/dockstore/client/cli/CheckUrlHelperFullIT.java

```diff
@@ -104,14 +110,67 @@ void settingVersionMetadata() throws Exception {
+     * injectable via the dockstore.yml (there's already an interface).
+     */
+    @Test
+    void testUpdateOpenData() {
+        CommonTestUtilities.cleanStatePrivate2(EXT.getTestSupport(), false, testingPostgres);
+        ApiClient client = getOpenAPIWebClient(USER_2_USERNAME, testingPostgres);
+        io.dockstore.openapi.client.api.EntriesApi
+            entriesApi = new io.dockstore.openapi.client.api.EntriesApi(client);
+        WorkflowsApi workflowsApi = new WorkflowsApi(client);
+        assertEquals(0, rowsWithPublicAccessibleData());
+        Workflow wdlWorkflow = workflowsApi.manualRegister(SourceControl.GITHUB.name(), "DockstoreTestUser2/hello-dockstore-workflow", "/Dockstore.wdl", "",
+            DescriptorLanguage.WDL.getShortName(), "");
+        wdlWorkflow = workflowsApi.refresh1(wdlWorkflow.getId(), true);
+        Workflow cwlWorkflow = workflowsApi.manualRegister(SourceControl.GITHUB.name(), "DockstoreTestUser2/hello-dockstore-workflow", "/Dockstore.cwl", "cwlworkflow",
+            DescriptorLanguage.CWL.getShortName(), "");
+        cwlWorkflow = workflowsApi.refresh1(cwlWorkflow.getId(), true);
+
+        // Clear publicaccessibletestparameterfile
+        testingPostgres.runUpdateStatement("update version_metadata set publicaccessibletestparameterfile = null");
+        // Confirm the above worked cleared out the language versions
+        wdlWorkflow = workflowsApi.getWorkflow(wdlWorkflow.getId(), null);
+        wdlWorkflow.getWorkflowVersions().forEach(wv -> assertNull(wv.getVersionMetadata().isPublicAccessibleTestParameterFile()));
+        cwlWorkflow = workflowsApi.getWorkflow(cwlWorkflow.getId(), null);
+        cwlWorkflow.getWorkflowVersions().forEach(wv -> assertNull(wv.getVersionMetadata().isPublicAccessibleTestParameterFile()));
+
+        final Integer processed = entriesApi.updateOpenData(Boolean.TRUE);
+        assertEquals(2, processed);
+
+        wdlWorkflow = workflowsApi.getWorkflow(wdlWorkflow.getId(), null);
+        wdlWorkflow.getWorkflowVersions().forEach(wv -> assertTrue(!wv.isValid() || wv.getVersionMetadata().isPublicAccessibleTestParameterFile(),
+            "No valid versions have file inputs, should be open"));
+
+        cwlWorkflow = workflowsApi.getWorkflow(cwlWorkflow.getId(), null);
+        final List<WorkflowVersion> validVersions =
+            cwlWorkflow.getWorkflowVersions().stream().filter(WorkflowVersion::isValid).toList();
+        assertEquals(2, validVersions.size());
+        assertTrue(validVersions.stream().anyMatch(wv -> wv.getVersionMetadata().isPublicAccessibleTestParameterFile()),
+            "One valid version has no file inputs, should be open");
+        assertTrue(validVersions.stream().anyMatch(wv -> !wv.getVersionMetadata().isPublicAccessibleTestParameterFile()),
+            "One valid version has file inputs and no test parameter file, should not be open");
```

Comment:
Same question

Reply:
Same answer :) . Resolved in upcoming push.